sql-plugin/src/main/scala/com/nvidia/spark/RebaseHelper.scala

```diff
@@ -37,7 +37,16 @@ object RebaseHelper extends Arm {
           }
         }
       }
-    } else if (dtype.isTimestampType) {
+    } else {
+      false
+    }
+  }
+
+  private[this] def isTimeRebaseNeeded(column: ColumnVector,
+      startDay: Int,
+      startTs: Long): Boolean = {
+    val dtype = column.getType
+    if (dtype.isTimestampType) {
```

Comment:
This would allow `TIMESTAMP_DAYS`.
```suggestion
    if (dtype.hasTimeResolution) {
```